scipy/optimize/tests/test__basinhopping.py

```diff
@@ -212,6 +212,7 @@ def test_all_minimizers(self):
                                niter=self.niter, disp=self.disp)
             assert_almost_equal(res.x, self.sol[i], self.tol)
 
+    @pytest.mark.fail_slow(40)
```

Comment:
For reference, the test runs this fast locally on an `i9-13900K`:

```
5.13s call     build-install/lib/python3.11/site-packages/scipy/optimize/tests/test__basinhopping.py::TestBasinHopping::test_all_nograd_minimizers
```

If I do Matt's suggested experiment of parametrization, the timings drop off pretty fast after that new method.

```
4.43s call     build-install/lib/python3.11/site-packages/scipy/optimize/tests/test__basinhopping.py::TestBasinHopping::test_all_nograd_minimizers[COBYQA]
0.15s call     build-install/lib/python3.11/site-packages/scipy/optimize/tests/test__basinhopping.py::TestBasinHopping::test_all_nograd_minimizers[COBYLA]
0.13s call     build-install/lib/python3.11/site-packages/scipy/optimize/tests/test__basinhopping.py::TestBasinHopping::test_all_nograd_minimizers[CG]
0.08s call     build-install/lib/python3.11/site-packages/scipy/optimize/tests/test__basinhopping.py::TestBasinHopping::test_all_nograd_minimizers[TNC]
0.07s call     build-install/lib/python3.11/site-packages/scipy/optimize/tests/test__basinhopping.py::TestBasinHopping::test_all_nograd_minimizers[BFGS]

```

If you did parametrize, I believe you could use `pytest.param` to apply the timing exception or another marker to just that one test case, so that other regressions don't sneak in. If you apply `xslow`, then seems like it would only make sense to apply it to the one case that way.

Reply:
I think the settings can be adjust so it doesn't take so long, though. The tolerance to pass the test is quite coarse. Can the termination settings be adjusted to avoid unnecessary iterations?